src/hardware/serialport/misc_util.h

```diff
@@ -60,7 +60,13 @@
 
 #include "enet/include/enet.h"
 
-enum SocketTypesE { SOCKET_TYPE_TCP = 0, SOCKET_TYPE_ENET, SOCKET_TYPE_COUNT };
+enum class SocketType {
+	Tcp  = 0,
+	Enet = 1,
+	Invalid,
```

Comment:
Should we not specify 2 as the value?

Reply:
Might be OK, but will let @FeralChild64 comment:

An enum _without_ any values tells me we're using the names in a "pure enum" sense, without ever needing to cast-down to numbers.

So when I read this, the first `= 0` and `= 1` tell me we're casting them down to numbers and (maybe) passing them into a C function.

But the bare 'Invalid' tells me we only use it as an enum, perhaps in an assert.

Reply:
I've fixed the values of `Tcp` and `Enet` due to the following code in `softmodem.cpp` (if I understand correctly, this is a modem emulation code):
```C++
const uint32_t requested_mode = ScanNumber(scanbuf);
const auto requested_type = static_cast<SocketType>(requested_mode);
```
The `Invalid` - well, I just want it to be the first invalid value. I'll add a comment there.

Reply:
> While the bare 'Invalid' tells me we only use it as an enum, perhaps in an assert (so it communicates to me that we never expect to cast `Invalid` down to a number).

Yeah, that's fine and a good way to think about it. But we're comparing values to Invalid in the code, so that's I thought we might want to specify the next value explicitly. But the maybe it's part of the standard that auto-assigned enum values have no holes.